### PR TITLE
Wrong uuid set to ReAuthOperation caller uuid.

### PR DESCRIPTION
**Pull request by sancar**
Added ownership tests for client

### CODE REVIEW

**Comment by sancar** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractMessageTask.java [hunk 1]
```diff
@@ -218,6 +213,7 @@ protected Connection findSendConnection() {
             return connection;
         }
 
+        String clientUuid = endpoint.getUuid();
```
Comment (ihsandemir) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractMessageTask.java]: after this change it seems that you can also remove the cached clientUuid.

File: hazelcast-client/src/test/java/com/hazelcast/client/ClientOwnershipTest.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
+package com.hazelcast.client;
```
Comment (ihsandemir) [inline hazelcast-client/src/test/java/com/hazelcast/client/ClientOwnershipTest.java]: may be you can add this case: client connected, cluster shutdown or all members shutdown, then start the cluster (or one member) again and check owner.

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractMessageTask.java (hunk 1 above)
Comment (sancar) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractMessageTask.java]: Do you mean this https://github.com/hazelcast/hazelcast/pull/9514/files#diff-62dfc5312563d71974e936132c44bd73L70 ?

Comment (ihsandemir) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractMessageTask.java]: ok, already removed :+1

File: hazelcast-client/src/test/java/com/hazelcast/client/ClientOwnershipTest.java (hunk 2 above)
Comment (sancar) [inline hazelcast-client/src/test/java/com/hazelcast/client/ClientOwnershipTest.java]: I added the test just now

**Pull request merged**